docs/architecture/adr-111-nil-mempool.md

```diff
@@ -0,0 +1,239 @@
+of the Cosmos SDK,
+and briefly discussed in the last bullet point above (see [SDK app-mempool][sdk-app-mempool] for further details of this model).
+
+In the app-mempool design in Cosmos SDK `v0.47.x`
+an unconfirmed transaction must be both in CometBFT's mempool for dissemination and
+in the app's mempool so the application can decide how to manage the mempool.
+The many advantages of this approach are beyond question. However it has some implications:
+
+* Having every transaction both in CometBFT and in the application is suboptimal in terms of memory.
+  Additonally, the app developer has to be careful
+  that the contents of both mempools does not diverge over time
+  (hence the crucial role `re-CheckTx` plays post-ABCI++).
+* The main reason for a transaction needing to be in CometBFT's mempool is
+  because the design in Cosmos SDK `v0.47.x` does not consider an application
+  that has its own means of disseminating transactions.
+  It reuses the peer to peer network underneath CometBFT reactors.
+* So, if an app has an ad-hoc design of how to disseminate transactions,
+  there is no point in having transactions in CometBFT's mempool.
+
+This proposal targets this kind of applications:
+those that have an ad-hoc (and likely more efficient) mechanism for transaction dissemination.
+We propose to introduce a `nil` mempool which will effectively act as a stubbed object
+internally:
+
+* it will reject any transaction being locally submitted or gossipped by a peer
+* when it is time to _reap_ (as it is currently called) the mempool the answer will always be empty
+* the application running on the proposer validator will add transactions it received
+  using the appchains's own mechanism via `PrepareProposal`.
+
+## Alternative Approaches
+
+These are the alternatives known to date:
+
+1. Keep the current model. Useful for basic apps, but clearly suboptimal for applications
+   with their own mechanism to disseminate transactions and particular performance requirements.
+2. Provide more efficient general-purpose mempool implementations.
+   This is an ongoing effort, but will take some time (and R&D effort) to come up with
+   advanced mechanisms -- likely highly configurable -- which then will have to be thoroughly tested.
+3. A similar approach to this one ([ADR110][adr-110]) whereby the application-specific
+   mechanism directly interacts with CometBFT via a newly defined gRPC interface.
+
+## Decision
+
+TBD
+
+## Detailed Design
+
+What this ADR proposes can already be achieved with an unmodified CometBFT since `v0.37.x`,
+albeit with a complex, poor UX.
+This core of this proposal is to make some internal changes so it is clear an simple for app developers,
+thus improving the UX.
+
+#### `nil` Mempool
+
+We propose a new mempool implementation, called `nil` Mempool, that effectively disables all mempool functionality
+within CometBFT.
+The `nil` Mempool implements the `Mempool` interface in a very simple manner:
+
```

Comment:
Did we confirm that all these returns on the `nil` mempool will not cause any side effects or undesirable behaviours ? From what I read the idea of going with this model is that would not required any extra changes except on the interface implementation level.

Reply:
To the extent that I crawled back all callsites by hand and convinced myself there'll be no extra changes in other parts of the code.
But, when implementing, we might find something unexpected that might require this ADR to be updated (as we usually do).